README.md

```diff
@@ -108,43 +108,94 @@ See [CONFIG.md](CONFIG.md) for details.
+
+## 1. IdentifiedDataSerializable Serialization
+For a faster serialization of objects, Hazelcast recommends to implement IdentifiedDataSerializable interface.
+
+Here is an example of an object implementing IdentifiedDataSerializable interface:
 
 ```javascript
-writeData(dataOutput) {
-    ...
-    dataOutput.writeInt(this.arrayOfCustomObjects);
-    this.arrayOfCustomObjects.forEach(function(element) {
-        dataOutput.writeObject(element);
-    });
-    ...
+function Address(street, zipCode, city, state) {
+    this.street = street;
+    this.zipCode = zipCode;
+    this.city = city;
+    this.state = state;
 }
 
-readData(dataInput) {
-    ...
-    var arrayOfCustomObjects = [];
-    var lenOfArray = dataInput.readInt();
-    for (i=0;i<lenOfArray;i++) {
-        arrayOfCustomObjects.push(dataInput.readObject());
+Address.prototype.getClassId = function () {
+    return 1;
+};
+
+Address.prototype.getFactoryId = function () {
+    return 1;
+};
+
+Address.prototype.writeData = function (objectDataOutput) {
+    objectDataOutput.writeUTF(this.street);
+    objectDataOutput.writeInt(this.zipCode);
+    objectDataOutput.writeUTF(this.city);
+    objectDataOutput.writeUTF(this.state);
+};
+
+Address.prototype.readData = function (objectDataInput) {
+    this.street = objectDataInput.readUTF();
+    this.zipCode = objectDataInput.readInt();
+    this.city = objectDataInput.readUTF();
+    this.state = objectDataInput.readUTF();
+};
+```
+
+IdentifiedDataSerializable uses `getClassId()` and `getFactoryId()` to reconstitute the object. To complete the implementation `IdentifiedDataSerializableFactory` should also be implemented and registered into `SerializationConfig` which can be accessed from `Config.serializationConfig`. The factory's responsibility is to return an instance of the right `IdentifiedDataSerializable` object, given the class id. 
+
+A sample `IdentifiedDataSerializableFactory` could be implemented as following:
+
+```javascript
+var myIdentifiedFactory = {
+    create: function (type) {
+        if (type === 1) {
+            return new Address();
+        }
     }
-    this.arrayOfCustomObjects = arrayOfCustomObjects;
-    ...
-}
+};
 ```
 
+The last step is to register the `IdentifiedDataSerializableFactory` to the `SerializationConfig`.
+
+```javascript
+var config = new Config.ClientConfig();
+config.serializationConfig.dataSerializableFactories[1] = myIdentifiedFactory;
+```
+
+So far this is the most efficient way of serialization that Hazelcast supports of the shelf.
```

Comment:
It is not right to say Identified is the most efficient. Portable is more efficient in terms of querying. They have some strengths over the other. That is why we maintain both of them.

Reply:
I will remove it.